### PR TITLE
classify now changes state when image is correctly classified

### PR DESCRIPTION
**Pull request by hakosx**
# Changes :gem:
classify event now changes state when image is correctly classifie
moved some code into models.py
updated test to test that nothing breaks

# Notes :book: :pencil2:
_Write a few notes about the changes made if necessary._

### CODE REVIEW

File: src/test/websocket_test.py [hunk 1]
```diff
@@ -24,16 +24,20 @@ def test_join_game_responds(client):
...
 def test_classification_correct(client):
     flask_client, ws_client = client
 
     assert ws_client.is_connected()
 
-    # ws_client.emit("joinGame", {})
+    ws_client.emit("joinGame", {})
 
+    r = ws_client.get_received()
+    print(r[0]["args"][0])
```
Comment (mahbx) [inline src/test/websocket_test.py:44]: do we need this print?

File: src/test/websocket_test.py [hunk 2]
```diff
@@ -24,16 +24,20 @@ def test_join_game_responds(client):
...
+    args = json.loads(r[0]["args"][0])
+    game_id = args["game_id"]
     path = "harambe.png"
```
Comment (mahbx) [inline src/test/websocket_test.py:47]: we need this file, store it in test/data?

File: src/test/websocket_test.py [hunk 3]
```diff
@@ -47,10 +51,18 @@ def test_classification_correct(client):
...
+        data = {"game_id": game_id, "time_left": 1}
+
+        ws_client.emit("classify", data, image.stream.read())
 
     r = ws_client.get_received()
 
-    assert r[1]["name"] == "prediction"
-    print(r[1])
-    assert False
+    print(r)
```
Comment (mahbx) [inline src/test/websocket_test.py:66]: do we need this print?

File: src/webapp/api.py [hunk 4]
```diff
@@ -127,16 +127,45 @@ def get_label(game_id):
     return json.dumps(data)
 
 
+def translate_probabilities(labels):
+    translation_dict = models.get_translation_dict()
```
Comment (mahbx) [inline src/webapp/api.py:134]: docstring missing

File: src/webapp/api.py [hunk 5]
```diff
@@ -127,16 +127,45 @@ def get_label(game_id):
...
+    return dict(
+        [(translation_dict[label], prob) for label, prob in labels.items()]
+    )
+
+
 @socketio.on("classify")
 def handle_classify(data, image):
```
Comment (mahbx) [inline src/webapp/api.py:141]: docstring :D

File: src/test/websocket_test.py [hunk 6]
```diff
@@ -23,19 +28,25 @@ def test_join_game_responds(client):
...
 
-    print(r)
-    assert r[0]["name"] == "message"
+    print("joingame event: ", r)
+    assert r[0]["name"] == "player_info"
 
 
 def test_classification_correct(client):
+    """
+        tests wether a player is able to join a game and submit a image for classification and get the result from the classification
```
Comment (andreacx) [inline src/test/websocket_test.py:37]: too long;)

File: src/webapp/api.py [hunk 7]
```diff
@@ -127,16 +127,54 @@ def get_label(game_id):
...
+        if opponent_done:
+            models.update_game_for_player(player_id, game_id, 1, "Done")
+            emit("round_over", room=game_id)
+
+    response = {
+        "certainty": translate_probabilities(prob_kv),
+        "guess": models.to_norwegian(best_guess),
+        "correctLabel": models.to_norwegian(correct_label),
+        "hasWon": has_won,
+    }
```
Comment (andreacx) [inline src/webapp/api.py:182]: sexy classify!!!

**Pull request merged**